msgcntr/messageforums-app/src/java/org/sakaiproject/tool/messageforums/DiscussionForumTool.java

```diff
@@ -3381,6 +3405,73 @@ else if(getUserId()==null&&this.forumManager.getAnonRole()==true){
     return decoTopic;
   }
 
+	public void setAnonymousManager(AnonymousManager anonymousManager)
+	{
+		this.anonymousManager = anonymousManager;
+	}
+
+	public boolean isAnonymousEnabled()
+	{
+		return ServerConfigurationService.getBoolean("msgcntr.forums.anonymous.enable", true);
+	}
+
+	public boolean isSiteHasAnonymousTopics()
+	{
+		return forumManager.isSiteHasAnonymousTopics(getSiteId());
+	}
+
+	/**
+	 * Determines whether the current user should see anonymous IDs in the context of the specified topic.
+	 * @param topic
+	 * @param true if the topic is anonymous and revealIDsToRoles is disabled or
+	 * revealIDsToRoles is enabled, but the user doesn't have the permission to identify users in this topic
+	 */
+	private boolean isUseAnonymousId(Topic topic)
+	{
+		if (topic == null)
+		{
+			throw new IllegalArgumentException("isUseAnonymousId invoked with null topic");
+		}
+
+		if (!isAnonymousEnabled())
+		{
+			return false;
+		}
+
+		// if topic.postAnonymous
+		//   if topic.revealIDsToRoles
+		//     return !uiPermissionsManager.isIdentifyAnonAuthors (anonymous only if they don't have permission to see identities)
+		//   return true (anonymous for all)
+		//  return false (not anonymous)
+
+		// Condenses to
+		return topic.getPostAnonymous() && (!topic.getRevealIDsToRoles() || !uiPermissionsManager.isIdentifyAnonAuthors(topic));
+	}
+
+	/**
+	 * Gets a userId -> anonymousID map containing entries for every author obtained from the 'messages' list.
+	 * Does so in a single query (unless anonymous mappings are missing, in which case they are created)
+	 * @param messages list of Messages
+	 */
+	private Map<String, String> getUserIdAnonIdMapForMessages(List messages)
```

Comment:
Will do. Didn't know you can pass regular Lists into methods that take Lists with generics!